tripy/tripy/frontend/module/batchnorm.py

```diff
@@ -0,0 +1,118 @@
+
+    where:
+        - :math:`\mu` is the precomputed running mean.
+        - :math:`\sigma^2` is the precomputed running variance.
+        - :math:`\gamma` and :math:`\beta` are learnable parameter vectors (wieight and bias).
+
+    This implementation supports 1D, 2D, and 3D inputs (e.g., time-series, images, and volumetric data).
+    Batch Normalization normalizes across the specified feature dimension (typically the second dimension in the input).
+
+    """
+
+    num_features: int
+    r"""The number of feature channels in the input tensor (the size of the second dimension)."""
+
+    dtype: datatype.dtype
+    r"""The data type used to perform the operation."""
+
+    eps: float
+    r""":math:`\epsilon` value added to the denominator to prevent division by zero during normalization."""
+
+    weight: Parameter
+    r"""The :math:`\gamma` parameter of shape :math:`[\text{num_features}]`."""
+
+    bias: Parameter
+    r"""The :math:`\beta` parameter of shape :math:`[\text{num_features}]`."""
+
+    running_mean: Parameter
+    r"""The running mean for the feature channels of shape :math:`[\text{num_features}]`."""
+
+    running_var: Parameter
+    r"""The running variance for the feature channels of shape :math:`[\text{num_features}]`."""
+
+    def __init__(self, num_features: int, dtype: datatype.dtype = datatype.float32, eps: float = 1e-5) -> None:
+        r"""
+        Args:
+            num_features: The number of feature channels in the input tensor (the size of the second dimension).
+            dtype: The data type to use for the weight, bias, running_mean and running_var parameters.
+            eps: :math:`\epsilon` value added to the denominator to prevent division by zero during normalization.
+
+        .. code-block:: python
+            :linenos:
+            :caption: Example
+
+            batch_norm = tp.BatchNorm(2)
+
+            input = tp.iota((1, 2, 1, 1))
+            output = batch_norm(input)
+        """
+        super().__init__()
+
+        self.num_features = num_features
+        self.eps = eps
+
+        # Initialize learnable parameters (scale and shift)
+        self.weight = DefaultParameter((num_features,), dtype=dtype)
+        self.bias = DefaultParameter((num_features,), dtype=dtype)
+
+        # Initialize running statistics (precomputed, not updated)
+        self.running_mean = DefaultParameter((num_features,), dtype=dtype)
+        self.running_var = DefaultParameter((num_features,), dtype=dtype)
+
+    def __call__(self, x: "tripy.Tensor") -> "tripy.Tensor":
+        r"""
+        Args:
+            x: The input tensor with shape :math:`(N, C, ...)`, where C is the feature dimension.
+
+        Returns:
+            A tensor of the same shape as the input.
+        """
+        from tripy.frontend.trace.ops.unary_elementwise import rsqrt
+        from tripy.frontend.trace.ops.reshape import reshape
+
+        x_shape = (1, -1, *([1] * (x.rank - 2)))
```

Comment:
Since we know the dimension, can we just use `self.num_features` instead of `-1`?

Reply:
done